res/apicurio-template.yml

```diff
@@ -0,0 +1,953 @@
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    name: postgresql
+    creationTimestamp: null
+    labels:
+      app: postgresql
+      template: apicurio-studio
+  spec:
+    strategy:
+      type: Recreate
+      recreateParams:
+        timeoutSeconds: 600
+      resources: {}
+      activeDeadlineSeconds: 21600
+    triggers:
+      - type: ImageChange
+        imageChangeParams:
+          automatic: true
+          containerNames:
+            - postgresql
+          from:
+            kind: ImageStreamTag
+            namespace: openshift
+            name: 'postgresql:9.5'
+      - type: ConfigChange
+    replicas: 1
+    test: false
+    selector:
+      name: postgresql
+    template:
+      metadata:
+        creationTimestamp: null
+        labels:
+          name: postgresql
+      spec:
+        volumes:
+          - name: postgresql-data
+            persistentVolumeClaim:
+              claimName: postgresql-data
+        containers:
+          - name: postgresql
+            image: centos/postgresql-95-centos7
+            ports:
+              - containerPort: 5432
+                protocol: TCP
+            env:
+              - name: POSTGRESQL_USER
+                valueFrom:
+                  secretKeyRef:
+                    name: postgresql
+                    key: database-user
+              - name: POSTGRESQL_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: postgresql
+                    key: database-password
+              - name: POSTGRESQL_DATABASE
+                value: ${DB_NAME}
+            resources:
+              limits:
+                memory: 512Mi
+            volumeMounts:
+              - name: postgresql-data
+                mountPath: /var/lib/pgsql/data
+            livenessProbe:
+              tcpSocket:
+                port: 5432
+              initialDelaySeconds: 30
+              timeoutSeconds: 1
+              periodSeconds: 10
+              successThreshold: 1
+              failureThreshold: 3
+            readinessProbe:
+              exec:
+                command:
+                  - /bin/sh
+                  - '-i'
+                  - '-c'
+                  - >-
+                    psql -h 127.0.0.1 -U $POSTGRESQL_USER -q -d
+                    $POSTGRESQL_DATABASE -c 'SELECT 1'
+              initialDelaySeconds: 5
+              timeoutSeconds: 1
+              periodSeconds: 10
+              successThreshold: 1
+              failureThreshold: 3
+            terminationMessagePath: /dev/termination-log
+            imagePullPolicy: Always
+            securityContext:
+              capabilities: {}
+              privileged: false
+        restartPolicy: Always
+        terminationGracePeriodSeconds: 30
+        dnsPolicy: ClusterFirst
+        securityContext: {}
+  status: {}
+# Keycloak Deployment Configuration
+# #################################
+- apiVersion: v1
+  kind: DeploymentConfig
```

Comment:
Do you know if we should be using a kubernetes "Deployment" here rather than a "DeploymentConfig"? @maleck13 @odra Anyone any idea what the differences are and if one is better than the other, or even if they serve the same purpose?

Reply:
What I tried to achieve is to leave the upstream template untouched while using it in the operator so we don't have to change the template at all.

We agreed in the past to use the upstream template if there is one, otherwise we would create/use base kubernetes data types (ingress, deployment, etc)

Reply:
Yeah OK, i thought you had created this tbh. If it's upstream then yeah, we should leave it as is.